cocotb/triggers.py

```diff
@@ -181,8 +178,12 @@ class _ParameterizedSingletonAndABC(ParametrizedSingleton, abc.ABCMeta):
 
 
 class ReadOnly(with_metaclass(_ParameterizedSingletonAndABC, GPITrigger)):
-    """Execution will resume when the readonly portion of the sim cycles is
-    reached.
+    """Fires when the current simulation timestep moves to the readonly phase.
+
+    The :any:`ReadOnly` phase is entered when the current timestep no longer has any further delta steps.
+    This should be a point where all the signal values are stable as there are no more RTL events scheduled for the timestep.
```

Comment:
Is there a way of making clearer that these two "should"s here do not mean an obligation of the user but rather that the simulator will behave like this? Maybe just use "will"?

Reply:
Copied verbatim from the rst that was there before.

"Will" is probably reasonable. Alternatively, we could leave documenting all of these timing events to a future PR, since I don't really have the VPI background (yet).

Reply:
I'd just use "will" and do more later if needed.